### PR TITLE
Logging image efficiency and test

### PR DESCRIPTION
**Pull request by Xierumeng**

### CODE REVIEW

File: modules/logger/logger.py [hunk 1]
```diff
@@ -99,15 +100,25 @@ def create(cls, name: str, enable_log_to_file: bool) -> "tuple[bool, Logger | No
             file_handler.setFormatter(formatter)
             logger.addHandler(file_handler)
 
-        return True, Logger(cls.__create_key, logger)
+            return True, Logger(
+                cls.__create_key, logger, pathlib.Path(log_directory_path, log_path)
+            )
+
+        return True, Logger(cls.__create_key, logger, None)
```
Comment (maxlou05) [inline modules/logger/logger.py:107]: Why is there 2 returns here?

**Review by maxlou05 — COMMENTED** [review]
Reviewed

File: modules/logger/logger.py (hunk 1 above)
Comment (Xierumeng) [reply modules/logger/logger.py:107]: `log_path` doesn't exist when `enable_log_to_file` is false.

**Review by maxlou05 — APPROVED** [review]
Approved

**Pull request merged**